Literal table

shakram02 opened the pull request:
- Remove ugly hex padding function

shakram02 committed:
remove hex padding

shakram02 committed:
prepare for lit-tab

shakram02 committed:
formatting

shakram02 committed:
Literal table skeleton

shakram02 committed:
Base table complete

shakram02 committed:
Literal table

shakram02 committed:
Add literal table

shakram02 committed:
Resolve conflicts after merging

shakram02 commented:
meh :-1: forced update ruined the history

shakram02 committed:
lazy regex

shakram02 committed:
ENCAPSULATION rules

shakram02 committed:
pass1 adaptation

shakram02 committed:
More compromise between handler and htme

shakram02 committed:
Fomratting! -> it's simple

shakram02 committed:
fix broken htme tests

shakram02 committed:
Pretty prints and literals

-   Reduce import verbosity by re-exporting
-   Impl Display for formats
-   Impl Display for iinstruction
-   Fix bug in literal table
-   Use `RawProg` in `filehandler` and `pass_one`
-   Literal creation and usage
-   Base in progress

shakram02 committed:
Fix pop bug, ignore htme test file

shakram02 committed:
Literal resolution and injection

oddcoder commented on src/bin/main.rs:
```diff
@@ -39,17 +39,20 @@ fn main() {
-        }
+
+    let asm_file = FileHandler::new(input);
+    let listing_info = sick_lib::pass_one::pass_one::pass_one(asm_file);
```
https://is.gd/ls0Oyu

oddcoder commented on Cargo.toml:
```diff
@@ -21,4 +21,5 @@ getopts = "*"
-lazy_static = "*"
\ No newline at end of file
+lazy_static = "*"
+parking_lot = "*"
```
why ?

oddcoder commented on src/lib/basic_types/base_table.rs:
```diff
@@ -0,0 +1,105 @@
+/// if a base is available for xtr
+
+use std::u32;
+use parking_lot::RwLock;
```
we dont have multi threading here on any level so explain, mutexes or what ever you used here introduce new level of bugs and there should always be better ways to solve it, pls just dont over do it

shakram02 replied:
static access needs locking

oddcoder commented on src/lib/basic_types/formats.rs:
```diff
@@ -7,8 +7,13 @@ pub enum Format {
     None,
 }
-
+use std::fmt;
```
this should go to the top of the file

oddcoder commented on src/lib/htme/raw_program.rs:
```diff
@@ -5,11 +5,12 @@ use basic_types::instruction::Instruction;
 use basic_types::operands::*;
 use htme::record_string::*;
 
+#[derive(Debug)]
```
@Youssefares ^_^ ur part

oddcoder commented on src/lib/pass_one/pass_one.rs:
```diff
@@ -1,11 +1,16 @@
-use basic_types::operands::*;
+use unit_or_pair::*;
+use instruction_set::is_action_directive;
+use parking_lot::RwLock;
```
fuck mutexes ...

oddcoder commented on src/lib/pass_one/pass_one.rs:
```diff
@@ -41,43 +49,151 @@ fn get_instruction_size (inst:&Instruction) -> i32 {
-        }
+
+pub fn pass_one(mut file: FileHandler) -> (HashMap<String, i32>, RawProgram) {
+
```
remove all those empty lines ... they make readablity hard

shakram02 replied:
no they don't, it's a convention to leave empty lines in code and between functions

shakram02 committed:
fix length calculation bug & Literal API improvement

shakram02 committed:
reduce import length

shakram02 committed:
i32/u32 confusion reduction

shakram02 committed:
Bugs everywhere

-   Fixed bug with pass_one in RESW
-   Fixed major bugs in `instruction.rs`
    - Format 4 flags
    - Mismatched flags
    - Default indirect and immediate flags
-   Fixed bug with RSUB definition
-   Added getters/setters for instruction format to set the flags
-   Integrate pass2 with pass1
-   Pass2 loop encapsulation
-   Pretty printing for operands
-   Fix corrupt unit tests

shakram02 commented:
this should also close #17
